flood_adapt/object_model/hazard/hazard.py

```diff
@@ -317,16 +317,20 @@ def preprocess_sfincs(
                     const_dir=self.event.attrs.wind.constant_direction.value,
                 )
 
-            # Add measures if included
-            if self.hazard_strategy.measures is not None:
-                for measure in self.hazard_strategy.measures:
-                    measure_path = base_path.joinpath(
-                        "input", "measures", measure.attrs.name
+        # Add floodwall if included
+        if self.hazard_strategy.measures is not None:
+            for measure in self.hazard_strategy.measures:
+                measure_path = base_path.joinpath(
+                    "input", "measures", measure.attrs.name
+                )
+                if measure.attrs.type == "floodwall":
+                    model.add_floodwall(
+                        floodwall=measure.attrs, measure_path=measure_path
+                    )
+                if measure.attrs.type == "green_infrastructure":
+                    model.add_green_infrastructure(
+                        green_infrastructure=measure.attrs, measure_path=measure_path
```

Comment:
This is missing one indent and will only write out the last SFINCS model for an event set
```suggestion
          # Add floodwall if included
          if self.hazard_strategy.measures is not None:
              for measure in self.hazard_strategy.measures:
                  measure_path = base_path.joinpath(
                      "input", "measures", measure.attrs.name
                  )
                  if measure.attrs.type == "floodwall":
                      model.add_floodwall(
                          floodwall=measure.attrs, measure_path=measure_path
                      )
                  if measure.attrs.type == "green_infrastructure":
                      model.add_green_infrastructure(
                          green_infrastructure=measure.attrs, measure_path=measure_path
```